### ISSUE TITLE
[Task]: Adjust auto-approval delay to take into account weekends

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons#15095]
### Description

We already have support for adjusting for weekends when calculating due dates - we want to implement that for auto-approval delay too.

### Acceptance Criteria

```[tasklist]
  ### Milestones/checkpoints
  - [ ] An auto-approval that would fall over the weekend is adjusted to be approved after the weekend instead
```

### Checks

- [X] If I have identified that the work is specific to a repository, I have removed "repository:addons-server" or "repository:addons-frontend"

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1149)

**Comment by eviljeff** [comment]
@abhn / @wagnerand can you confirm the exact criteria here: https://mozilla-hub.atlassian.net/browse/AOP-137 says "between 4pm on a Friday and 10am on the following Sunday" - this is the submission date/time, yes?  i.e. the criteria isn't based on when the _approval_ would happen, but the submission?

Sceneries for submission, assuming a 24 hour delay (configurable):
- submitted at 15:59 Friday; so auto approved ~15:59 Saturday?
- submitted at 16:01 Friday; so auto approved ~15:59 Monday?  or ~10:01 Sunday (42 hours later)? or ~10:01 Monday (24+42 hours later)?
- submitted at 10:01 Sunday; so auto approved ~10:01 Monday?

Also, these times are UTC or some other timezone?

**Comment by diox** [comment]
It's not explicit in this issue, but this is only about the auto-approval delay for the first listed version, right ?

Also, now that [this delay is configurable](https://github.com/mozilla/addons/issues/15061) if the delay is set to something higher than 24 hours, say 72 hours, could that cause a new listed add-on to have to wait 5 days in total ? Is that a problem ?

**Comment by wagnerand** [comment]
@diox the single soft-coded delay would probably be replaced by this more flexible approach. However, I could see it still having value for incidents or surges, where we might have to up the delay even during the week. 

I am interesting why we chose Friday 4pm to Sunday (all day?). More consistent to me would be either Saturday and Sunday all day, or Friday 4pm - Monday 7am (exact hour tbd).

I'll let Abhi reply to the other questions.

**Comment by abhn** [comment]
(I'm fading on my memory on the original motivation but will try to answer as I now see things. Given the configurable delay and the fact that we keep add-ons in content queue after auto-approval, the context for this ticket has changed a bit).

TLDR of what would make most sense to me is if the queue just didn't take weekends into account. Assuming a 24 hour delay - 

- submitted at 12.34pm on Friday, auto-approval on 12.34 on Monday
- submitted at 16.01 on Friday, auto-approval on 16.01 on Monday
- submitted anytime between 00.00 Saturday to 00.00 Monday, auto-approval 00.00 Tuesday. 

If we thinks this is a better approach, I'll also update my original jira ticket. 

To answer some of the questions from @eviljeff (may be invalidated by my answer above)

> "between 4pm on a Friday and 10am on the following Sunday" - this is the submission date/time, yes?
Yep, it was the submission time, under the assumption of a 24 hour static auto-approval delay. 

> submitted at 15:59 Friday; so auto approved ~15:59 Saturday? 
Yep, as we generally clear the content queues at the end of the workday on friday, that scenario shouldn't happen without the add-on going through the content review.

> submitted at 16:01 Friday; so auto approved ~15:59 Monday? or ~10:01 Sunday (42 hours later)? or ~10:01 Monday (24+42 hours later)?
I think this part I didn't think through. Ideally, 16.01 on Monday. The weekend time isn't part of the configured delay in auto-approval. The idea was that the queue acts like the weekend didn't exist and Friday was a day behind Monday.

> submitted at 10:01 Sunday; so auto approved ~10:01 Monday?
Like the previous point, I think it would be 00.00 Tuesday

> Also, these times are UTC or some other timezone?
These are CE(S)T, although if we ignore weekends and just add carry-over the delay from Friday to Monday, that wouldn't matter much.

**Comment by eviljeff** [comment]
Okay, ignoring entire days (Saturday and Sunday) is the easy path too so :+1:

**Comment by diox** [comment]
We discussed in triage to keep the delay configurable but translate it to business days (essentially skipping weekends as pointed out by @eviljeff above).

**Comment by diox** [comment]
Currently not a high-priority as the delay is set by Operations and no longer hardcoded to 24 hours. Content review is also making that less necessary.
